packages/search-in-workspace/src/browser/search-layout-migrations.ts

```diff
@@ -0,0 +1,53 @@
+/********************************************************************************
+ * Copyright (C) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from '@theia/core/shared/inversify';
+import { ApplicationShellLayoutMigration, WidgetDescription, ApplicationShellLayoutMigrationContext } from '@theia/core/lib/browser/shell/shell-layout-restorer';
+import { SearchInWorkspaceWidget } from './search-in-workspace-widget';
+import { SEARCH_VIEW_CONTAINER_ID, SEARCH_VIEW_CONTAINER_TITLE_OPTIONS } from './search-in-workspace-factory';
+
+@injectable()
+export class SearchLayoutVersion3Migration implements ApplicationShellLayoutMigration {
+    readonly layoutVersion = 6.0;
+    onWillInflateWidget(desc: WidgetDescription, { parent }: ApplicationShellLayoutMigrationContext): WidgetDescription | undefined {
+        if (desc.constructionOptions.factoryId === SearchInWorkspaceWidget.ID && !parent) {
+            return {
```

Comment:
There seems to be an issue related to the new codicon icon for the search widget:

![image](https://user-images.githubusercontent.com/4377073/132725521-3f4aefb0-5977-4fd9-af7c-123cde3ed18a.png)

Simply add the following in between the `if` and `return`:

```ts
if (typeof desc.innerWidgetState === 'string') {
    desc.innerWidgetState = desc.innerWidgetState.replace(/search-in-workspace-tab-icon/g, codicon('search'));
}
```

Reply:
Hi @msujew, why not to change [this line](https://github.com/EstherPerelman/theia/blob/94f37b3812e961d076a4369fab0b07a968630cd1/packages/search-in-workspace/src/browser/search-in-workspace-factory.ts#L29)
to ` iconClass: codicon('search')`

Reply:
Right, that probably makes more sense 👍